stdlib/LinearAlgebra/src/structuredbroadcast.jl

```diff
@@ -64,7 +64,39 @@ find_bidiagonal() = throw(ArgumentError("could not find Bidiagonal within broadc
 find_bidiagonal(a::Bidiagonal, rest...) = a
 find_bidiagonal(bc::Broadcast.Broadcasted, rest...) = find_bidiagonal(find_bidiagonal(bc.args...), rest...)
 find_bidiagonal(x, rest...) = find_bidiagonal(rest...)
+
+find_uplo(a::Bidiagonal) = a.uplo
+find_uplo(a::Char) = a
+find_uplo(a::Number) = Nothing
+find_uplo(a) = Nothing
+
+function find_uplo(bc::Broadcast.Broadcasted, rest...)
+
+    left = find_uplo(bc.args[1])
+    right = find_uplo(bc.args[2])
+
+    tmp = (left, right)
+    myuplo = Nothing
+    for i = 1:length(tmp)
+        if typeof(tmp[i]) == Char
```

Comment:
It'd be a bit more inference-friendly to use `if tmp[I] isa Char` in places like this... but often seeing something like this in an if statement like this is a signal that you want to be using dispatch instead!